plonky2/src/gates/poseidon2_risc0_babybear.rs

```diff
@@ -45,8 +45,16 @@ const NON_ROUTED_WIRES_PER_OP: usize =
     SPONGE_CAPACITY + SPONGE_WIDTH * (N_FULL_ROUNDS_TOTAL - 1) + N_PARTIAL_ROUNDS;
 
 impl<F: RichField + HasExtension<D>, const D: usize> Poseidon2R0BabyBearGate<F, D> {
-    pub fn new() -> Self {
-        Self::new_from_config(&CircuitConfig::standard_recursion_config_bb_wide())
+    pub fn new_with_num_ops(num_ops: usize) -> Self {
+        Self {
+            num_ops,
+            _phantom: PhantomData,
+        }
+    }
+
+    fn num_ops(config: &CircuitConfig) -> usize {
+        let wires_per_op = ROUTED_WIRES_PER_OP + NON_ROUTED_WIRES_PER_OP;
+        (config.num_wires / wires_per_op).min(config.num_routed_wires / ROUTED_WIRES_PER_OP)
     }
```

Comment:
This function can now be used to compute `num_ops` inside the `new_from_config()` constructor below (credits to @DDT92).